### DISCUSSION TITLE
Components

### DISCUSSION THREAD

**Discussion by danieludokike** [reactive-python/reactpy-django#164] [Question] [answered]
1. Do the components replace the Django view or it's callable on Django's view?
2. Can I create a base template with all the statics in terms of SPA and the styles that take place in the .html components?
3. Does the reactpy have hooks as does react I want to get the full docs as this sounds like a game changer.

**Comment by Archmonger** [comment, marked as answer]
1. You need [at least one Django view](https://reactive-python.github.io/reactpy-django/get-started/register-view/) in order to put your component(s) within. Outside of that, you technically can recreate all of your existing HTML templates within ReactPy.
2. Yes, you can define your static files in a base Django template, and then use the [`extends` template tag](https://docs.djangoproject.com/en/4.2/ref/templates/language/#template-inheritance).
3. Yes, we currently have [replicated all common ReactJS hooks](https://reactpy.dev/docs/reference/hooks-api.html).
